templates/cmd/controller/main.go.tpl

```diff
@@ -86,14 +92,30 @@ func main() {
 		os.Exit(1)
 	}
 
+	defaultNamesSpaces := make(map[string]ctrlrtcache.Config, 0)
+	if ackCfg.WatchNamespace != "" {
+		// If use has specified multiple namespaces, we need to watch all of them
+		// by adding them to the defaultNamesSpaces map.
+		namespaces := strings.Split(ackCfg.WatchNamespace, ",")
+		for _, namespace := range namespaces {
+			defaultNamesSpaces[namespace] = ctrlrtcache.Config{}
+		}
+	}
 	mgr, err := ctrlrt.NewManager(ctrlrt.GetConfigOrDie(), ctrlrt.Options{
-		Scheme:                  scheme,
-		Port:                    port,
-		Host:                    host,
-		MetricsBindAddress:      ackCfg.MetricsAddr,
+		Scheme: scheme,
+		Cache: ctrlrtcache.Options{
+			Scheme:            scheme,
+			DefaultNamespaces: defaultNamesSpaces,
+		},
```

Comment:
I'm not sure if this is the ideal approach. I'm not certain but I think we'd want to use `cache.MultiNamespacedCacheBuilder` in a more dynamic way. See example [here](https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/#configuring-cluster-scoped-operators-with-multinamespacedcachebuilder)

Reply:
That was possible in `0.15.3` which is the version sdk-operator still using, however it was deprecated in 0.16.0 (https://github.com/kubernetes-sigs/controller-runtime/pull/2423) and a newer granular configurations were possible with:
- https://github.com/kubernetes-sigs/controller-runtime/pull/2421/files
- https://github.com/kubernetes-sigs/controller-runtime/pull/2539/files

Reply:
AHH okay cool. I'm still not up to speed on `0.16.x`